### ISSUE TITLE
Stop server when a plugin cannot be loaded

### ISSUE THREAD

**Issue by SOF3** [pmmp/PocketMine-MP#3080]
This includes loading failure due to incompatible API, unresolved dependencies and other reasons known by PocketMine.

**Comment by robske110** [comment]
Maybe an alternative would be to give plugins the ability to indicate that they think the server can still continue starting (or that the server should not continue starting).

**Comment by SOF3** [comment]
AllAPILoader/plugin.yml:

```yml
api: [1.0.0]
i-think-the-server-can-still-start-and-ignore-my-incompatible-api: true
```

**Comment by SOF3** [comment]
We are talking about cases where we are sure the plugin can't be loaded. It is not up to the plugin to decide whether the server can run without it, but up to the server owner.

**Comment by Muqsit** [comment]
Besides, there's a `plugin_list.yml` on the dev build which shifts the decision making ability on the server owner.

**Comment by dktapps** [comment]
Implemented by #4951
